many/src/List.js

```diff
@@ -1,100 +1,58 @@
+  return {Props: xs.of({color: hexColor, width: randomWidth})};
+}
 
-  let mutableLastId = 0;
+function makeItems(count) {
+  const items = [];
 
-  function createNewItem(props) {
-    const id = mutableLastId++;
-    const sinks = itemFn(props, id);
-    return {id, DOM: sinks.DOM.remember(), Remove: sinks.Remove};
+  for (let i = 0; i < count; i += 1) {
+    items.push(createRandomItemProps());
   }
 
-  const addItemReducer$ = action$
-    .filter(a => a.type === 'ADD_ITEM')
-    .map(action => {
-      const amount = action.payload;
-      let newItems = [];
-      for (let i = 0; i < amount; i++) {
-        newItems.push(createNewItem(createRandomItemProps()));
-      }
-      return function addItemReducer(listItems) {
-        return listItems.concat(newItems);
-      };
-    });
-
-  const removeItemReducer$ = action$
-    .filter(a => a.type === 'REMOVE_ITEM')
-    .map(action => function removeItemReducer(listItems) {
-      return listItems.filter(item => item.id !== action.payload);
-    });
+  return items;
+}
 
-  const initialState = [createNewItem({color: 'red', width: 300})]
+function intent(DOM) {
+  return xs.merge(
+    DOM.select('.add-one-btn').events('click')
+      .map(() => makeItems(1)),
 
-  return xs.merge(addItemReducer$, removeItemReducer$)
-    .fold((listItems, reducer) => reducer(listItems), initialState);
+    DOM.select('.add-many-btn').events('click')
+      .map(() => makeItems(1000))
+  );
 }
 
 function view(items$) {
-  const addButtons = div('.addButtons', [
-    button('.add-one-btn', 'Add New Item'),
-    button('.add-many-btn', 'Add Many Items')
-  ]);
-
-  return items$.map(items => {
-    const itemVNodeStreamsByKey = items.map(item =>
-      item.DOM.map(vnode => {
-        vnode.key = item.id; return vnode;
-      })
+  return Collection.pluck(items$, 'DOM')
+    .map(items =>
+      div('.list', [
+        div('.addButtons', [
+          button('.add-one-btn', 'Add New Item'),
+          button('.add-many-btn', 'Add Many Items')
+        ]),
+
+        div(items)
+      ])
     );
-    return xs.combine(...itemVNodeStreamsByKey)
-      .map(vnodes => div('.list', [addButtons].concat(vnodes)));
-  }).flatten();
-}
-
-function makeItemWrapper(DOM) {
-  return function itemWrapper(props, id) {
-    const item = isolate(Item)({DOM, Props: xs.of(props)});
-    return {
-      DOM: item.DOM,
-      Remove: item.Remove.mapTo(id)
-    }
-  }
 }
 
 function List(sources) {
-  const proxyItemRemove$ = xs.create();
-  const action$ = intent(sources.DOM, proxyItemRemove$);
-  const itemWrapper = makeItemWrapper(sources.DOM);
-  const items$ = model(action$, itemWrapper);
-  const itemRemove$ = items$
-    .map(items => xs.merge(...items.map(item => item.Remove)))
-    .flatten();
-  proxyItemRemove$.imitate(itemRemove$);
+  const addItem$ = intent(sources.DOM)
+    .startWith({Props: xs.of({color: 'red', width: 300})});
+
+  const items$ = Collection(Item, sources, addItem$, 'Remove');
```

Comment:
Is that just for the sake of being explicit? Because I'm not sure if you could achieve any additional functionality with this API over the removeSinkName style. I think there's something I'm missing.